src/hooks/use-request.ts

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import type { AxiosError, AxiosRequestConfig, CancelTokenSource } from 'axios'
+import axios from 'axios'
+
+interface UseRequestConfig<TResponse, TError = unknown> extends AxiosRequestConfig {
+  onSuccess?: (data: TResponse) => void
+  onError?: (error: AxiosError<TError>) => void
+}
+
+interface UseRequestResult<TResponse, TError = unknown> {
+  data: TResponse | null
```

Comment:
Maybe

```suggestion
interface UseRequestResult<TResponseData, TError = unknown> {
  data: TResponseData | null
```

, because it's easy to confuse it with `AxiosResponse` type. 
As I understand it, `data` here is `axiosResponse.data`, not `axiosResponse` itself.